packages/firebase_auth/firebase_auth/example/lib/signin_page.dart

```diff
@@ -195,33 +195,43 @@ class _EmailLinkSignInSectionState extends State<_EmailLinkSignInSection>
   @override
   void didChangeAppLifecycleState(AppLifecycleState state) async {
     if (state == AppLifecycleState.resumed) {
-      final Uri link = await _retrieveDynamicLink();
-
-      if (link != null) {
-        final FirebaseUser user = (await _auth.signInWithEmailAndLink(
-          email: _userEmail,
-          link: link.toString(),
-        ))
-            .user;
-
-        if (user != null) {
-          _userID = user.uid;
-          _success = true;
-        } else {
-          _success = false;
-        }
-      } else {
-        _success = false;
+      final PendingDynamicLinkData data =
+          await FirebaseDynamicLinks.instance.getInitialLink();
+      if (data?.link != null) {
+        handleLink(data?.link);
```

Comment:
`handleLink` is `async`. Does it need to be `await`ed?

Reply:
Hrm, I don't know on that one. It's working as is, but I can try awaiting it in the morning.

Reply:
Doesn't need to be since handleLink returns a void (rather than a Future).